repomgr/repomgr.go

```diff
@@ -705,14 +709,18 @@ func (rm *RepoManager) BatchWrite(ctx context.Context, user models.Uid, writes [
 			}
 
 			if rm.hydrateRecords {
-				op.Record = c.Value.Val
+				op.Record = ltd.Val
 			}
 
 			ops = append(ops, op)
 		case w.RepoApplyWrites_Update != nil:
 			u := w.RepoApplyWrites_Update
 
-			cc, err := r.PutRecord(ctx, u.Collection+"/"+u.Rkey, u.Value.Val)
+			var ltd lexutil.LexiconTypeDecoder
+			if err = ltd.UnmarshalJSON(*u.Value); err != nil {
```

Comment:
Is this adding a marshalling step in repomanager where we didn't have one before? Wondering if there will be performance impacts on this cause this is critical path on relay write-path iirc.